src/google/protobuf/map.h

```diff
@@ -176,14 +176,12 @@ class Map {
 
     template<class NodeType>
     void destroy(NodeType* p) {
-      if (arena_ == NULL) p->~NodeType();
+      p->~NodeType();
     }
 #else
     void construct(pointer p, const_reference t) { new (p) value_type(t); }
 
-    void destroy(pointer p) {
-      if (arena_ == NULL) p->~value_type();
-    }
+    void destroy(pointer p) { p->~value_type(); }
```

Comment:
No, we cannot. The caller may depend on the destructor to do clean-ups which is expected to happen right now. E.g., after destroy() the caller may call constructor() on the same memory block right away.